web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificateTypesFacadeServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.common.support.modules.registry.IntygModuleRegistry;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.CertificateTypeInfoDTO;
+import se.inera.intyg.schemas.contract.InvalidPersonNummerException;
+import se.inera.intyg.schemas.contract.Personnummer;
+import se.inera.intyg.webcert.web.service.facade.GetCertificateTypesFacadeService;
+import se.inera.intyg.webcert.web.web.controller.api.dto.IntygModuleDTO;
+import se.inera.intyg.webcert.web.web.util.resourcelinks.ResourceLinkHelper;
+
+@Service
+public class GetCertificateTypesFacadeServiceImpl implements GetCertificateTypesFacadeService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GetCertificateTypesFacadeServiceImpl.class);
+
+    private final IntygModuleRegistry intygModuleRegistry;
+    private final ResourceLinkHelper resourceLinkHelper;
+
+
+    @Autowired
+    public GetCertificateTypesFacadeServiceImpl(IntygModuleRegistry intygModuleRegistry, ResourceLinkHelper resourceLinkHelper) {
+        this.intygModuleRegistry = intygModuleRegistry;
+        this.resourceLinkHelper = resourceLinkHelper;
+    }
+
+    @Override
+    public List<CertificateTypeInfoDTO> get(String patientId) {
+        final var certificateModuleList = getCertificateModuleList(patientId);
+        return certificateModuleList.stream().map(this::convertModuleToTypeInfo).collect(Collectors.toList());
+    }
+
+    private CertificateTypeInfoDTO convertModuleToTypeInfo(IntygModuleDTO module) {
+        final var certificateTypeInfo = new CertificateTypeInfoDTO();
+        certificateTypeInfo.setId(module.getId());
+        certificateTypeInfo.setLabel(module.getLabel());
+        certificateTypeInfo.setDescription(module.getDescription());
+        certificateTypeInfo.setDetailedDescription(module.getDetailedDescription());
+        certificateTypeInfo.setIssuerTypeId(module.getIssuerTypeId());
+        certificateTypeInfo.setLinks(module.getLinks());
+        return certificateTypeInfo;
+    }
+
+    private List<IntygModuleDTO> getCertificateModuleList(String patientId) {
+        try {
+            final var intygModules = intygModuleRegistry.listAllModules();
+
+            final var intygModuleDTOs = intygModules.stream()
+                .map(IntygModuleDTO::new)
+                .collect(Collectors.toList());
+
+            final var personnummer = createPnr(patientId);
+            resourceLinkHelper.decorateIntygModuleWithValidActionLinks(intygModuleDTOs, personnummer);
+
+            return intygModuleDTOs;
+
+        } catch (InvalidPersonNummerException e) {
+            LOG.error(e.getMessage());
+            return Collections.emptyList();
```

Comment:
Should this error just return an empty list. The web client will not be aware that something went wrong and the user will just be presented by an empty set of types.